File: intermediate/final/prog/subrect.py
#!/usr/bin/python3
import sys
import math

if __name__ == "__main__":
	N = 10
	if len(sys.argv) > 1:
		N = int(sys.argv[1])
	for k in range(0,N):
		nums = [int(x) for x in input().split()]
		m = nums[0]
		n = nums[1]
		idx = 2
		arr = [[0]*10 for _ in range(10)]
		for i in range(m):
			for j in range(n):
				arr[i][j] = nums[idx]
				idx += 1
		max_sum = -1 * sys.maxsize;
		# Complete Search each dxw rectangles
		for i in range(m):
			for j in range(n):
				for d in range(1,m+1):
					for w in range(1,n+1):
						if i+d <= m and j+w <= n:
							sum = 0
							for r in range(i,i+d):
								for c in range(j,j+w):
									sum += arr[r][c]
							max_sum = max(max_sum, sum)
		print(max_sum)

File: intermediate/final/prog/subrect.java
import java.io.*;
import java.util.*;

public class subrect {
	public static void main(String[] args) throws IOException {
		int N = args.length > 0 ? Integer.parseInt(args[0]) : 10;
		Scanner input = new Scanner(System.in);
		for(int k = 0; k < N && input.hasNextLine(); ++k) {
			int[][] arr = new int[10][10];
			int m = input.nextInt();
			int n = input.nextInt();
			for(int i = 0; i < m; ++i)
				for(int j = 0; j < n; ++j)
					arr[i][j] = input.nextInt();
			input.nextLine();
			int max_sum = Integer.MIN_VALUE;
			// Complete Search each dxw rectangles
			for(int i = 0; i < m; ++i) {
				for(int j = 0; j < n; ++j) {
					for(int d = 1; d <= m; ++d) {
						for(int w = 1; w <= n; ++w) {
							if(i+d <= m && j+w <= n) {
								int sum = 0;
								for(int r = i; r < i+d; ++r)
									for(int c = j; c < j+w; ++c)
										sum += arr[r][c];
								max_sum = Math.max(max_sum, sum);
							}
						}
					}
				}
			}
			System.out.println(max_sum);
		}
	}
}
